wire-runtime/src/main/java/com/squareup/wire/RuntimeMessageAdapter.java

```diff
@@ -44,7 +42,7 @@
   private final Class<M> messageType;
   private final Class<Builder<M>> builderType;
   private final Constructor<Builder<M>> builderCopyConstructor;
-  private final TagMap<RuntimeFieldBinding> fieldBindingMap;
+  private final Map<Integer, TagBinding<M, Builder<M>>> tagBindings;
```

Comment:
Not currently using `TagMap` 'cause it's awkward to create a derived TagMap. May revisit later; or potentially we don't need TagMap anymore because it's optimizing an uncommon case. (We should only have 1 TagMap in memory per message type in the common case.)